k8s/serviceaccount.go

```diff
@@ -0,0 +1,38 @@
+package k8s
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+)
+
+func UpdateServiceAccountFromYaml(k8sClient *kubernetes.Clientset, serviceAccountYaml string) error {
+	serviceAccount := v1.ServiceAccount{}
+
+	decoder := yamlutil.NewYAMLToJSONDecoder(bytes.NewReader([]byte(serviceAccountYaml)))
+	err := decoder.Decode(&serviceAccount)
+	if err != nil {
+		return err
+	}
+	for retries := 0; retries <= 5; retries++ {
+		_, err = k8sClient.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Create(&serviceAccount)
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				if _, err = k8sClient.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Update(&serviceAccount); err != nil {
+					logrus.Debugf("Failed to update ServiceAccount, retrying..")
+					time.Sleep(time.Second * 5)
```

Comment:
Can we just break from the loop when there is no error? This way you don't have to repeat the sleep in both conditions, and do it only when the execution was not successful